Unexpected EOF - findDatastore

Firstly thanks for writing this provider!

I've run into an issue however when trying to provision machine(s) into a Datastore which is in a folder.

For example: my datastores are arranged like Such:

```
Datacenter
  Development Datastores
    DVS01
  Staging Datastores
     SDS01
     SDS02
  Production Datastores
    PDS01
    PDS02
```

Getting the following error:

* unexpected EOF
Error applying plan:

5 error(s) occurred:

* unexpected EOF
* unexpected EOF
* unexpected EOF
* unexpected EOF
* unexpected EOF

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
panic: runtime error: invalid memory address or nil pointer dereference
2015/07/15 10:30:21 terraform-provider-vsphere: [signal 0xb code=0x1 addr=0x20 pc=0x44d7a3]
2015/07/15 10:30:21 terraform-provider-vsphere: 
2015/07/15 10:30:21 terraform-provider-vsphere: goroutine 81 [running]:
2015/07/15 10:30:21 terraform-provider-vsphere: github.com/rakutentech/terraform-provider-vsphere/vsphere.findDatastore(0xc2081401a0, 0xc2092c3ae0, 0xc208450780, 0xc20aa34380, 0xc2081d5ea0, 0x19, 0x0, 0x
0, 0x0)
2015/07/15 10:30:21 terraform-provider-vsphere:         /home/ubuntu/go/src/github.com/rakutentech/terraform-provider-vsphere/vsphere/virtual_machine.go:397 +0x283
2015/07/15 10:30:21 terraform-provider-vsphere: github.com/rakutentech/terraform-provider-vsphere/vsphere.(*virtualMachine).deployVirtualMachine(0xc2080e0b00, 0xc2081401a0, 0x0, 0x0)
2015/07/15 10:30:21 terraform-provider-vsphere:         /home/ubuntu/go/src/github.com/rakutentech/terraform-provider-vsphere/vsphere/virtual_machine.go:268 +0x23e2
2015/07/15 10:30:21 terraform-provider-vsphere: github.com/rakutentech/terraform-provider-vsphere/vsphere.resourceVSphereVirtualMachineCreate(0xc20807d2c0, 0xeb1880, 0xc2081401a0, 0x0, 0x0)
2015/07/15 10:30:21 terraform-provider-vsphere:         /home/ubuntu/go/src/github.com/rakutentech/terraform-provider-vsphere/vsphere/resource_vsphere_virtual_machine.go:270 +0x1b61
2015/07/15 10:30:21 terraform-provider-vsphere: github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc20812cc40, 0xc2081315f0, 0xc2081d7490, 0xeb1880, 0xc2081401a0, 0x1, 0x0, 0x0)
2015/07/15 10:30:21 terraform-provider-vsphere:         /home/ubuntu/go/src/github.com/hashicorp/terraform/helper/schema/resource.go:145 +0x2c1
2015/07/15 10:30:21 terraform-provider-vsphere: github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc208130270, 0xc2081cdcc0, 0xc2081315f0, 0xc2081d7490, 0x9be240, 0x0, 0x0)
2015/07/15 10:30:21 terraform-provider-vsphere:         /home/ubuntu/go/src/github.com/hashicorp/terraform/helper/schema/provider.go:155 +0x1da
2015/07/15 10:30:21 terraform-provider-vsphere: github.com/hashicorp/terraform/rpc.(*ResourceProviderServer).Apply(0xc2081e8cc0, 0xc2081d5c40, 0xc2081d7910, 0x0, 0x0)
2015/07/15 10:30:21 terraform-provider-vsphere:         /home/ubuntu/go/src/github.com/hashicorp/terraform/rpc/resource_provider.go:319 +0x7f
2015/07/15 10:30:21 terraform-provider-vsphere: reflect.Value.call(0xbf2fa0, 0xe80840, 0x13, 0xeccfb0, 0x4, 0xc20c4f9f28, 0x3, 0x3, 0x0, 0x0, ...)
2015/07/15 10:30:21 terraform-provider-vsphere:         /usr/local/go/src/reflect/value.go:419 +0x10e5
2015/07/15 10:30:21 terraform-provider-vsphere: reflect.Value.Call(0xbf2fa0, 0xe80840, 0x13, 0xc20c4f9f28, 0x3, 0x3, 0x0, 0x0, 0x0)
2015/07/15 10:30:21 terraform-provider-vsphere:         /usr/local/go/src/reflect/value.go:296 +0xbc
2015/07/15 10:30:21 terraform-provider-vsphere: net/rpc.(*service).call(0xc20812ccc0, 0xc20812cc80, 0xc2081ff558, 0xc208104000, 0xc2082194a0, 0x9be1e0, 0xc2081d5c40, 0x16, 0x9be240, 0xc2081d7910, ...)
2015/07/15 10:30:21 terraform-provider-vsphere:         /usr/local/go/src/net/rpc/server.go:382 +0x1f7
2015/07/15 10:30:21 terraform-provider-vsphere: created by net/rpc.(*Server).ServeCodec
2015/07/15 10:30:21 terraform-provider-vsphere:         /usr/local/go/src/net/rpc/server.go:476 +0x44a
2015/07/15 10:30:21 terraform-provider-vsphere:

--- a/vsphere/virtual_machine.go
+++ b/vsphere/virtual_machine.go
@@ -265,9 +265,13 @@ func (vm *virtualMachine) deployVirtualMachine(c *govmomi.Client) error {
 			return err
 		}
 	} else {
-		datastore, err = findDatastore(c, dcFolders, template, resourcePool, vm.datastore)
+		datastore, err = finder.Datastore(context.TODO(), vm.datastore)
 		if err != nil {
-			return err
+			// TODO: datastore cluster support in govmomi finder function
+			datastore, err = findDatastore(c, dcFolders, template, resourcePool, vm.datastore)
+			if err != nil {
+				return err
+			}
 		}
 	}
 	log.Printf("[DEBUG] datastore: %#v", datastore)

